core/src/main/java/org/infinispan/metrics/impl/CounterTrackerImpl.java

```diff
@@ -0,0 +1,29 @@
+package org.infinispan.metrics.impl;
+
+import java.util.Objects;
+
+import org.infinispan.commons.stat.CounterTracker;
+
+import io.micrometer.core.instrument.Counter;
+
+/**
+ * The {@link CounterTracker} that uses {@link Counter}.
+ */
+public class CounterTrackerImpl implements CounterTracker {
```

Comment:
Why do we need the `CounterTracker` interface? Is it to just abstract away from `Micrometer` in case we change the metrics implementation again?

Reply:
Yes, all interfaces are abstractions from micrometer classes.